File: CS/XtraTabWithButtons/BaseTabHandlerDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraTab.ViewInfo;
using DevExpress.XtraTab;
using System.Windows.Forms;
using DevExpress.Utils;
using DevExpress.Utils.Controls;

namespace XtraTabWithButtons
{
    public class BaseTabHandlerDescendant : BaseTabHandler
    {
        public BaseTabHandlerDescendant(IXtraTab tabControl)
            : base(tabControl)
        {
            
        }

        protected override bool OnMouseMove(MouseEventArgs e)
        {
            bool result = base.OnMouseMove(e);
            DXMouseEventArgs ee = DXMouseEventArgs.GetMouseArgs(e);
            if (ViewInfo != null)
            {
                BaseTabPageViewInfoDescendant infodes = ViewInfo.SelectedTabPageViewInfo as BaseTabPageViewInfoDescendant;
                infodes.PagePanel.ProcessEvent(new ProcessEventEventArgs(EventType.MouseMove, ee));
            }
            return result;
        }
        protected override bool OnMouseDown(MouseEventArgs e)
        {
            bool result = base.OnMouseDown(e);
            DXMouseEventArgs ee = DXMouseEventArgs.GetMouseArgs(e);
            
            BaseTabPageViewInfoDescendant infodes = ViewInfo.SelectedTabPageViewInfo as BaseTabPageViewInfoDescendant;
            if(infodes.PagePanel.Bounds.Contains(e.Location))
                infodes.PagePanel.ProcessEvent(new ProcessEventEventArgs(EventType.MouseDown, ee));
            return result;
        }

        protected override bool OnMouseUp(MouseEventArgs e)
        {
            bool result = base.OnMouseUp(e);
            DXMouseEventArgs ee = DXMouseEventArgs.GetMouseArgs(e);
            BaseTabPageViewInfoDescendant infodes = ViewInfo.SelectedTabPageViewInfo as BaseTabPageViewInfoDescendant;
            infodes.PagePanel.ProcessEvent(new ProcessEventEventArgs(EventType.MouseUp, ee));
            return result;
        }

        protected override void OnMouseLeave(EventArgs e)
        {
            base.OnMouseLeave(e);
            if (ViewInfo != null)
            {
                BaseTabPageViewInfoDescendant infodes = ViewInfo.SelectedTabPageViewInfo as BaseTabPageViewInfoDescendant;
                infodes.PagePanel.ProcessEvent(new ProcessEventEventArgs(EventType.MouseLeave, EventArgs.Empty));
            }
        }
    }
}

File: VB/XtraTabWithButtons/BaseTabHandlerDescendant.vb
Imports System
Imports DevExpress.XtraTab.ViewInfo
Imports DevExpress.XtraTab
Imports System.Windows.Forms
Imports DevExpress.Utils
Imports DevExpress.Utils.Controls

Namespace XtraTabWithButtons

    Public Class BaseTabHandlerDescendant
        Inherits BaseTabHandler

        Public Sub New(ByVal tabControl As IXtraTab)
            MyBase.New(tabControl)
        End Sub

        Protected Overrides Function OnMouseMove(ByVal e As MouseEventArgs) As Boolean
            Dim result As Boolean = MyBase.OnMouseMove(e)
            Dim ee As DXMouseEventArgs = DXMouseEventArgs.GetMouseArgs(e)
            If ViewInfo IsNot Nothing Then
                Dim infodes As BaseTabPageViewInfoDescendant = TryCast(ViewInfo.SelectedTabPageViewInfo, BaseTabPageViewInfoDescendant)
                infodes.PagePanel.ProcessEvent(New ProcessEventEventArgs(EventType.MouseMove, ee))
            End If

            Return result
        End Function

        Protected Overrides Function OnMouseDown(ByVal e As MouseEventArgs) As Boolean
            Dim result As Boolean = MyBase.OnMouseDown(e)
            Dim ee As DXMouseEventArgs = DXMouseEventArgs.GetMouseArgs(e)
            Dim infodes As BaseTabPageViewInfoDescendant = TryCast(ViewInfo.SelectedTabPageViewInfo, BaseTabPageViewInfoDescendant)
            If infodes.PagePanel.Bounds.Contains(e.Location) Then infodes.PagePanel.ProcessEvent(New ProcessEventEventArgs(EventType.MouseDown, ee))
            Return result
        End Function

        Protected Overrides Function OnMouseUp(ByVal e As MouseEventArgs) As Boolean
            Dim result As Boolean = MyBase.OnMouseUp(e)
            Dim ee As DXMouseEventArgs = DXMouseEventArgs.GetMouseArgs(e)
            Dim infodes As BaseTabPageViewInfoDescendant = TryCast(ViewInfo.SelectedTabPageViewInfo, BaseTabPageViewInfoDescendant)
            infodes.PagePanel.ProcessEvent(New ProcessEventEventArgs(EventType.MouseUp, ee))
            Return result
        End Function

        Protected Overrides Sub OnMouseLeave(ByVal e As EventArgs)
            MyBase.OnMouseLeave(e)
            If ViewInfo IsNot Nothing Then
                Dim infodes As BaseTabPageViewInfoDescendant = TryCast(ViewInfo.SelectedTabPageViewInfo, BaseTabPageViewInfoDescendant)
                infodes.PagePanel.ProcessEvent(New ProcessEventEventArgs(EventType.MouseLeave, EventArgs.Empty))
            End If
        End Sub
    End Class
End Namespace
